cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProfileHttpHandler.java

```diff
@@ -278,6 +278,8 @@ private void writeProfile(ProfileId profileId, FullHttpRequest request)
     } catch (JsonSyntaxException e) {
       throw new BadRequestException("Unable to parse request body. Please make sure it is valid JSON", e);
     }
+    String totalProcessingCpusLabel = getTotalProcessingCpusLabel(profileCreateRequest.getProvisioner());
```

Comment:
Why are you doing this on write only? Won't this have issues with upgrades?

Reply:
I don't think it will have any issue, if old profile is imported , then the label will be generated for it. right?

Reply:
It's not about importing, it's about instance upgrades